rezasm-source/rezasm-core/src/parser/line.rs

```diff
@@ -53,16 +56,34 @@ impl Line {
         }
 
         let mut arguments: Vec<ArgumentType> = Vec::new();
-        for argument in args_out {
-            arguments.push(match argument.get_target(word_size) {
-                Ok(arg) => arg,
-                Err(error) => return Err(error),
-            });
-        }
+        let instruction_retrieved =
+            match instruction_registry::get_instruction(instruction, args_out.len()) {
+                Ok(instruction) => instruction,
+                Err(e) => return Err(e),
+            };
 
-        let matching_instruction = get_instruction(instruction, &arguments)?;
+        if instruction_retrieved.get_types().len() != args_out.len() {
```

Comment:
We may want to make a different error that is more descriptive of the situation e.g., ParserError::InvalidInstructionArgumentCount(u8, u8) where the first u8 is the provided argument count and the second u8 is the expected number of arguments.